### PR TITLE
pythonPackages.infix: init at 1.2

### PR DESCRIPTION
**Pull request by pasunboneleve**
###### Motivation for this change

New package.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by worldofpeace** [comment]
Top of the `README.md` says
> This project is an unmaintained experiment. Don't use it for production code.

So I don't see a reason for this to be in nixpkgs.

**Comment by dmvianna** [comment]
I intend to merge `pycategories` to nixpkgs and this is a dependency. I know it is unmaintained, but it works and we use it. 

https://gitlab.com/danielhones/pycategories

-- 
Daniel Vianna

**Comment by worldofpeace** [comment]
> I intend to merge `pycategories` to nixpkgs and this is a dependency. I know it is unmaintained, but it works and we use it.

Ahh now I understand. Also note that it's totally acceptable to have one pr with multiple packages, i.e adding this and `pycategories`.

However I'd really think having this package globally available isn't a good idea. You can't submit issues upstream or submit prs (maintained fork would be needed). Especially when the author vehemently directs you not to even use it.

So what you can do is add it privately for `pycategories`, like
```nix
...

let

infix = callPackage ./infix.nix {};

in

buildPythonPackage {
...
```

File: pkgs/development/python-modules/infix/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+buildPythonPackage rec {
+  pname = "infix";
+  version = "1.2";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "a1bfdcf875bc072f41e426d0673f2e3017750743bb90cc725fffb292eb09648c";
+  };
+
+  doCheck = true;
```
Comment (worldofpeace) [inline pkgs/development/python-modules/infix/default.nix]:
```suggestion
  # No Tests
  doCheck = false;
```

I don't see any tests.

**Comment by dmvianna** [comment]
Cool, let's close it then and I'll have it as a private package to `pycategories`.

**Pull request closed without merging**